lib/galaxy/managers/datasets.py

```diff
@@ -684,6 +683,20 @@ def deserialize_metadatum(self, dataset_assoc, key, val, **context):
         # ...?
         return unwrapped_val
 
+    def deserialize_datatype(self, item, key, val, **context):
+        if not item.datatype.allow_datatype_change:
+            raise exceptions.RequestParameterInvalidException("The current datatype does not allow datatype changes.")
+        if not self.app.datatypes_registry.get_datatype_by_extension(val):
+            raise exceptions.RequestParameterInvalidException("The target datatype does not exist.")
+        if not self.app.datatypes_registry.get_datatype_by_extension(val).allow_datatype_change:
+            raise exceptions.RequestParameterInvalidException("The target datatype does not allow datatype changes.")
+        if not DatasetAssociationManager(self.app).ok_to_edit_metadata(item.dataset_id):
+            raise Exception("Dataset metadata could not be updated because it is used as input or output of a running job.")
```

Comment:
```suggestion
            raise RequestParameterInvalidException("Dataset metadata could not be updated because it is used as input or output of a running job.")
```
This isn't a server error, I think this falls into the invalid request parameter domain

Reply:
I previously suggested the inverse :laughing: . I suppose it's border-line, Galaxy could put the request in a queue instead of returning an error. Also, the dataset may be in use by another user with whom it is shared.

Reply:
I agree about the queue, but I think this is nothing admins should see as errors in the logs, which would happen with bare Exceptions, right ?